### DISCUSSION TITLE
Question about uefi memtest86+'s boot opitons

### DISCUSSION THREAD

**Discussion by yoshinoyatoko** [memtest86plus/memtest86plus#262] [General]
Hi
I have a question about boot options' transfer when using uefi memtest86+ in a shell environment.
For example:
I compiled a uefi verion, and run it in a uefi shell environment directly.
So, per this case, i couldn't transfer boot options to memtest86+, right?

Should i provide syslinux.cfg in the same directory?
Will memtest.efi search syslinux.cfg and parse it automatically?

Thanks

**Comment by debrouxl** [comment]
> I compiled a uefi verion, and run it in a uefi shell environment directly.
> So, per this case, i couldn't transfer boot options to memtest86+, right?

Right.

> Should i provide syslinux.cfg in the same directory?
> Will memtest.efi search syslinux.cfg and parse it automatically?

memtest.efi does not currently attempt to read any configuration file from a filesystem using the UEFI boot services. However, we know that memtest86+ should eventually be made to do that, though I think that configuration options should be made accessible through both the CLI and a configuration file, so that non-UEFI computers, which will keep being around for many years, aren't left in the cold.

Also, note that since you're building the binary yourself, not being able to pass configuration through the CLI or a file can easily be worked around by changing some defaults in the configuration variables from https://github.com/memtest86plus/memtest86plus/blob/main/app/config.c or the list of tests in https://github.com/memtest86plus/memtest86plus/blob/main/tests/tests.c , for instance :)

**Comment by martinwhitaker** [comment]
> Hi I have a question about boot options' transfer when using uefi memtest86+ in a shell environment. For example: I compiled a uefi verion, and run it in a uefi shell environment directly. So, per this case, i couldn't transfer boot options to memtest86+, right?

That depends on your BIOS. If it provides a way to set the UEFI image load options, you can use that to pass the boot options. The binary data in the load options is interpreted as a string of 16-bit wide characters, with character codes 1 to 127 being interpreted as the corresponding ASCII character and any other character code being treated as a string terminator. This matches the way boot options are passed to the Linux kernel when it is booted directly via its EFI stub.

**Comment by yoshinoyatoko** [comment]
Got it
Thank you!
